R/TuningInstanceMath.R

```diff
@@ -0,0 +1,125 @@
+      super$initialize(
+        id = "func_value",
+        range = c(-Inf, Inf),
+        minimize = TRUE
+      )
+    },
+
+    # evaluate test function (ignore "prediction" argument) based on
+    # hyperparameter values
+    score_internal = function(prediction, learner, ...) {
+
+      v = learner$param_set$values
+      # extract input for the function
+      x = unlist(v[names(v) != "budget"])
+      # sample epsilon error based on budget
+      eps = rgamma(1, shape = 1, scale = 1/v$budget)
+      # FIXME: how should we scale the error?
+      globalOptTests::goTest(x, self$test_fun) + eps
+    }
+  )
+)
+
+# Modified learner class, now responsible for defining the domain of the math
+# function
+# tfun: character name of valid mathematical function
+# budget is currently treated like a normal learner hyperparameter, that adds
+# error to the function value
+LearnerMathTestFun = R6Class(
+  "LearnerMathTestFun",
+  inherit = LearnerRegr,
+
+  public = list(
+    initialize = function(test_fun) {
+      # get the upper and lower boundaries of each dimension of the test func
+      b = globalOptTests::getDefaultBounds(test_fun)
+      # get the domain dimensions of the test function
+      k = globalOptTests::getProblemDimen(test_fun)
+      # create a new parameter set based on the received func properties
+      ps = Map(
+        function(i) ParamDbl$new(paste0("x", i), lower = b$lower[i], upper = b$upper[i]),
+        seq_len(k)
+      )
+      ps = ParamSet$new(ps)
+      # add budget parameter 
+      ps$add(ParamDbl$new("budget", lower = 1, upper = 100, tags = "budget"))
+
+      # init superclass
+      super$initialize(
+        id = "regr.mtf",
+        feature_types = c("logical", "integer", "numeric", "character", "factor", "ordered"),
+        predict_types = "response",
+        param_set = ps,
+        properties = "missings"
+      )
+    },
+
+    train_internal = function(task) {
+      # let's not change the budget DURING training - discuss!
+      #n = length(task$row_roles$use)
+      #self$param_set$values$budget = n / length(task$row_ids)
+
+      return(list())
+    },
+
+    predict_internal = function(task) {
+      response = rep(99, task$nrow)
+      PredictionRegr$new(task, response = response)
+    }
+  )
+)
+
+# TODO: multi crit with ZDT and DTLZ family
+# A tuning instance used for running tuners on mathematical test functions
+# function_name: character of a valid mathematical function
+# terminator: mlr3tuning terminator object responsible for termination of a 
+# tuning run
+TuningInstanceMath = R6Class(
+  "TuningInstanceMath",
+  inherit = TuningInstance,
+
+  public = list(
+    initialize = function(function_name, terminator) {
+      # init class instances for function domain and evaluation
+      m_measure = MeasureMathTestFun$new(function_name)
+      m_learner = LearnerMathTestFun$new(function_name)
+
+      # init TuningInstance
+      inst = super$initialize(
+        tsk("boston_housing"), 
```

Comment:
If I get it right, this is just carried around and serves no purpose?
Can we create a stupid minimal task here?